### PR TITLE
Attempt to fix flakyness of path traversal test

### PR DESCRIPTION
**Pull request by pcapriotti**
Instead of generating `String` values, which might be invalid for the file system encoding in some locales, generate bytestrings directy and encode them as paths.

I hope this fixes a flakyness in the path traversal tests reported by @jschaul.

## Checklist

 - [x] The **PR Title** explains the impact of the change.
 - [x] The **PR description** provides context as to why the change should occur and what the code contributes to that effect. This could also be a link to a JIRA ticket or a Github issue, if there is one.
 - [x] **changelog.d** contains the following bits of information ([details](https://github.com/wireapp/wire-server/blob/develop/docs/developer/changelog.md)):
   - [x] A file with the changelog entry in one or more suitable sub-sections. The sub-sections are marked by directories inside `changelog.d`.

### CODE REVIEW

File: services/federator/test/unit/Test/Federator/Monitor.hs
Comment (stephen-smith) [inline services/federator/test/unit/Test/Federator/Monitor.hs:419]: Probably need to exclude `0` / ASCII `NUL`, too right?

Comment (pcapriotti) [reply services/federator/test/unit/Test/Federator/Monitor.hs:419]: I'm not 100% sure, but I don't think so. It seems the roundtripping encoding used by GHC can handle arbitrary bytes. A 0 byte should just become a zero character, which is a valid code point, and roundtrip back to a 0 byte.

Comment (stephen-smith) [reply services/federator/test/unit/Test/Federator/Monitor.hs:419]: ?  Then, what does Char(s)? does `0x30 :: Word8` map to and vice-versa?  Also, whatever GHC roundtripping is happening should handle `0x2F` / `'/'`, but we still exclude it...

If this is going through some other form of filtering to make sure the Characters are valid in path components, why do we need to exclude something that maps to `/`?  If this is the only such filtering / processing, then we need to exclude stuff that outputs ASCII `NUL`.

It didn't look like there was any additional processing before this was used to access to the filesystem, though...

Comment (stephen-smith) [reply services/federator/test/unit/Test/Federator/Monitor.hs:419]: > `0x30 :: Word8` is mapped to unicode code point `\u0030`, i.e. `0`. Not sure why that would be special.

Then, shouldn't `0 :: Word8` be mapped to `'\u0000' :: Char` (aka ASCII NUL), which isn't a valid character in a file path, under most encodings.

**Review by stephen-smith — APPROVED** [review]

File: services/federator/test/unit/Test/Federator/Monitor.hs
Comment (pcapriotti) [reply services/federator/test/unit/Test/Federator/Monitor.hs:419]: > > Isn't that what I wrote two comments ago?
> 
> You wrote "A 0 byte should just become a zero character" which I interpreted as `0 :: Word8` maps to `'0' :: Char`. 

Oh, I see the confusion now. Sorry about that.

> I'm fine with leaving it as is if the test stays reliable. I think the filtering of `'/'` paired with the `useAsCString` just had me in the mindset that we needed the byte string to be a valid path component on Linux / POSIX, as `'/'` and NUL are the two banned bytes. (`':'` is also problematic in places, because of PATH-like things.)

Alright, thanks. Let's leave it for now then. I'll still wait for confirmation from @jschaul before merging, since I think he's the only one who's actually witnessed a failure so far.

**Review by jschaul — APPROVED** [review]
I have problems myself reproducing this, as it doesn't always happen, and annoyingly I am unable to get the tests to fail when only running the tests in question.

Both outputs below are from `develop`, not this branch: The top pane is output from running `make` (without further arguments), which eventually runs all tests including the federator ones. This *sometimes* exhibits the problem, and *sometimes* not. The bottom pane is a manual run of the federator tests. I ran the bottom command (with and without the quickcheck seed) 100 times in a loop and it never failed. I'm not sure if there's something that increases the likelihood for this to fail when I run all unit tests instead of just the federator ones.

traversal

But I ran `make` a few times on this branch, and I still get some failures in other packages (`hscim`, `types-common`) sometimes; but not in federator anymore. So this change seems to be useful.

I'll post the offending remaining sometimes-failing tests elsewhere to be addressed in another PR.

**Pull request merged**
